src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -65,9 +67,64 @@ public AgentSettings LoadSettings()
             return settings;
         }
 
+        private void checkAgentRootDirectorySecure()
+        {
+            Trace.Info(nameof(checkAgentRootDirectorySecure));
+
+            try
+            {
+                string rootDirPath = HostContext.GetDirectory(WellKnownDirectory.Root);
+
+                if (!String.IsNullOrEmpty(rootDirPath))
+                {
+                    // Get info about root folder
+                    DirectoryInfo dirInfo = new DirectoryInfo(rootDirPath);
+
+                    // Get directory access control list 
+                    DirectorySecurity directorySecurityInfo = dirInfo.GetAccessControl();
+                    AuthorizationRuleCollection dirAccessRules = directorySecurityInfo.GetAccessRules(true, true, typeof(NTAccount));
+
+
+                    // Get identity reference of the BUILTIN\Users group
+                    IdentityReference bulitInUsersGroup = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null).Translate(typeof(NTAccount));
+
+                    // Check if BUILTIN\Users group have modify/write rights for the agent root folder
+                    List<FileSystemAccessRule> potentiallyInsecureRules = (from rule in dirAccessRules.OfType<FileSystemAccessRule>().AsParallel()
```

Comment:
I'd like to suggest to shorten it a little bit. Does it make sense here?
```
List<FileSystemAccessRule> potentiallyInsecureRules = dirAccessRules.OfType<FileSystemAccessRule>()
                        .AsParallel()
                        .Where(rule =>
                            rule.IdentityReference == bulitInUsersGroup && rule.FileSystemRights.HasFlag(FileSystemRights.Write) || rule.FileSystemRights.HasFlag(FileSystemRights.Modify))
                        .ToList();
```

Reply:
Thanks for the suggestion! I implemented it.